### PR TITLE
refactor(MergeMapTo): clarify 'any' types in MergeMapToSubscriber

### PR DESCRIPTION
**Pull request by saneyuki**
This just intends to clarify & make strictly types.

### CODE REVIEW

File: src/operator/mergeMapTo.ts [hunk 1]
```diff
@@ -112,7 +112,7 @@ export class MergeMapToSubscriber<T, I, R> extends OuterSubscriber<T, I> {
     }
   }
 
-  private _innerSub(ish: any,
+  private _innerSub(ish: SubscribableOrPromise<I>,
```
Comment (david-driscoll) [inline src/operator/mergeMapTo.ts]: This is technically `ObservableInput<I>` because you could pass in an array or iterable.

Comment (saneyuki) [inline src/operator/mergeMapTo.ts]: @david-driscoll 

Okay. I fixed in https://github.com/ReactiveX/rxjs/pull/1749/commits/1492043d2630c378cb0869ed27268b4eff5d945b

**Comment by blesh** [comment]
@saneyuki there's a linting error: https://travis-ci.org/ReactiveX/rxjs/builds/136332318#L445

**Comment by saneyuki** [comment]
fixed the linting error.

**Comment by kwonoj** [comment]
Change looks good to me. I'll check in this around tomorrow if there's other suggestions around.

**Pull request merged**